### PR TITLE
XRT-573 Support M2M CU and copybo command in new KDS

### PR DESCRIPTION
**Pull request by mamin506**

### CODE REVIEW

File: src/runtime_src/core/common/drv/kds_core.c [hunk 1]
```diff
@@ -577,6 +578,14 @@ int kds_add_context(struct kds_sched *kds, struct kds_client *client,
 			return -EINVAL;
 		}
 		++client->virt_cu_ref;
+		/* a special handling for m2m cu :( */
```
Comment (maxzhen) [inline src/runtime_src/core/common/drv/kds_core.c:581]: should this special handling be done when the client reserves virtual CU for the first time only?

File: src/runtime_src/core/common/drv/kds_core.c [hunk 2]
```diff
@@ -601,6 +611,14 @@ int kds_del_context(struct kds_sched *kds, struct kds_client *client,
 			return -EINVAL;
 		}
 		--client->virt_cu_ref;
+		/* a special handling for m2m cu :( */
```
Comment (maxzhen) [inline src/runtime_src/core/common/drv/kds_core.c:614]: see above comment, done for the last release?

File: src/runtime_src/core/common/drv/kds_core.c [hunk 3]
```diff
@@ -617,6 +635,9 @@ insert_cu(struct kds_cu_mgmt *cu_mgmt, int i, struct xrt_cu *xcu)
 {
 	cu_mgmt->xcus[i] = xcu;
 	xcu->info.cu_idx = i;
+	/* m2m cu */
+	if (xcu->info.intr_id == 128)
```
Comment (maxzhen) [inline src/runtime_src/core/common/drv/kds_core.c:639]: need a more descriptive name for const number 128

**Review by maxzhen — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/common/drv/kds_core.c (hunk 1 above)
Comment (mamin506) [reply src/runtime_src/core/common/drv/kds_core.c:580]: Sure. we can.

**Review by xdavidz — APPROVED** [review]

**Review by maxzhen — APPROVED** [review]

**Pull request merged**